api4/saml.go

```diff
@@ -273,14 +273,24 @@ func resetAuthDataToEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 	var params *ResetAuthDataParams
 	jsonErr := json.NewDecoder(r.Body).Decode(&params)
 	if jsonErr != nil {
-		c.Err = model.NewAppError("resetAuthDataToEmail", "model.utils.decode_json.app_error", nil, jsonErr.Error(), http.StatusBadRequest)
+		c.Err = model.NewAppError("resetAuthDataToEmail", "model.utils.decode_json.app_error", nil, "", http.StatusBadRequest).Wrap(jsonErr)
 		return
 	}
 	numAffected, appErr := c.App.ResetSamlAuthDataToEmail(params.IncludeDeleted, params.DryRun, params.SpecifiedUserIDs)
 	if appErr != nil {
 		c.Err = appErr
 		return
 	}
-	b, _ := json.Marshal(map[string]any{"num_affected": numAffected})
-	w.Write(b)
+
+	n := struct {
+		NumAffected int `json:"num_affected"`
+	}{
+		NumAffected: numAffected,
+	}
+
+	b, _ := json.Marshal(n)
+	_, err := w.Write(b)
```

Comment:
Usually we do, `json.NewEncoder(w).Encode(...)` to do it in a single step. 0/5 as long as we log the error.

Reply:
Right, will change that.

Reply:
Fixed